lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -79,7 +79,7 @@ struct AnnoTarget {
       llvm_unreachable("Unknown port instruction");
     }
     if (op->getNumResults() == 0)
-      return FIRRTLType();
+      return {FIRRTLType(), 0};
     return op->getResult(0).getType().cast<FIRRTLType>().getSubTypeByFieldID(
         fieldIdx);
   }
```

Comment:
I realize that this change makes this function is horrible, but I have a much larger change that reorganizes the `fieldID` calculation which will clean this up.  It will also fix the following bug:
```
circuit Foo: %[[{"class":"test","target":"~Foo|Foo>w[0][1]"}]]
  module Foo:
    wire w: UInt<1>[1][8]
 ```
 ```
 ./test5.fir:3:5: error: the 2-th token of [0][1] is out of range in the vector
    wire w: UInt<1>[1][8]
    ^
  firrtl.circuit "Foo"   {
    firrtl.module @Foo() {
      %w = firrtl.wire  : !firrtl.vector<vector<uint<1>, 1>, 8>
    }
  }
  ```